addons/sale_crm/models/sale_order.py

```diff
@@ -13,3 +13,23 @@ class SaleOrder(models.Model):
 
     def action_confirm(self):
         return super(SaleOrder, self.with_context({k:v for k,v in self._context.items() if k != 'default_tag_ids'})).action_confirm()
```

Comment:
call the method updating the code hereabove (instead of in action_cancel)

Reply:
Actually earlier it was not needed because `_message_post_after_hook` will be called ultimately after calling `action_confirm` but now we are using the context so we need to call our method here.